generator/util/strFunctions.py

```diff
@@ -149,114 +308,302 @@ def singular(name):
+    description of that type.
+
+    :param given_type: the type we want to wrap, e.g. 'array', 'enum',...
+    :param element:
+    :param hack: special case which can be used for 'element' type
+    :return: string describing the type and the element
+    """
+    if given_type == 'array':
         return 'consisting of an array of \\primtype{' + element + '}'
-    elif name == 'enum':
+    elif given_type == 'enum':
         element_name = texify(element)
         return 'of type \\primtype{' + element_name + '}'
-    elif name == 'element':
+    elif given_type == 'element':
         if hack:
             return 'of type \\' + element
         else:
             return wrap_token(element)
-    elif name == 'lo_element':
+    elif given_type == 'lo_element':
         return wrap_token(element)
-    elif name == 'inline_lo_element':
+    elif given_type == 'inline_lo_element':
         return 'TO DO: add type'
     else:
-        return 'of type \\primtype{' + name + '}'
+        return 'of type \\primtype{' + given_type + '}'
 
 
 def wrap_section(name, add_class=True, add_extended=False):
+    """
+    'Wrap' a section
+
+    TODO - a section of what???
+
+    :param name:
+    :param add_class:
+    :param add_extended:
+    :return:
+
+    e.g. "Cat" -> "\sec{cat-class}"
+    """
     if add_class:
         return '\\sec{' + make_class(name, add_extended) + '}'
     else:
         return '\\sec{' + name + '}'
 
 
 def make_class(name, add_extended=False):
+    """
+    Add '-class' to end of a lower-case string
+
+    :param name: the string to take in, and convert to lower-case
+    :param add_extended: if True, prepend result with 'extended-'
+    :return: the string made as a class
+
+    .. code-block:: default
+
+        make_class('Irenaeus') -> "irenaeus-class"
+        make_class('Irenaeus", True) -> "extended-irenaeus-class"
+        
+    """
     if add_extended:
         return 'extended-' + name.lower() + '-class'
     else:
         return name.lower() + '-class'
 
 
 def wrap_enum(name):
+    """
+    'Wrap' an enum
+
+    :param name: the enum to wrap
+    :return: the wrapped form of the enum
+
+    e.g. 'cat' -> '\\primtype{cat}'
+
+    TODO when, how and why would this be used? It's used in the
+    validation rules code, but at the moment I don't understand that.
+    """
 #    return '\\primtype{' + lower_first(name) + '}'
     return '\\primtype{' + name + '}'
 
+# End of latex specific functions
+##################################################
+
 
 def get_sid_refs(refs):
+    """
+    This function is specifically designed for when an attribute can point to
+    one or more element types and formatting lines in text referring to the
+    element(s).
+
+    :param refs: a string which is a comma-separated sequence of element names
+    :return: a tuple where the first is a string representing a textual list
+    of the elements and the second is a string representing a combined type.
```

Comment:
indent the second line of the text describing the `return` statement so it will display properly when run through Sphinx.